File: passwordKalkulator_C++/word.cpp
#include "word.h"
#include "exception.h"

Word::Word(QString s) : startingData(4), str(s) {}
Word::~Word() {}
QString Word::getWord() const{
    return str;
}
QString Word::getData() const{
    return str;
}
void Word::validate(){
    //tolgo spazi
    str.replace( " ", "" );
    //controllo che non ci siano numeri o caratteri speciali
    if(str.toStdString().find_first_of("0123456789") != std::string::npos || str.toStdString().find_first_of("\\|&<>`~!£$%/()=?^'[]{}*+-#@°ç,._;:\"") != std::string::npos){//se ci sono numeri
        throw Exception("Parola non valida,\nsono consentite solo lettere");
    }
}
bool Word::checkLength() {
    if(str.length()<3)
        throw Exception("I campi utilizzati per calcolare\nla password, devono essere\nlunghi almeno 3 caratteri");
    return true;
}

File: passwordKalculator_Java/StartingData/Word.java
package StartingData;
import Exception.MyException;
public class Word extends StartingData{
	private String str;
	
    public Word(String s){
		super(4);
		str=s;
	}
	public String getWord(){
		return str;
	}
	@Override
	public String getData(){
		return str;
	}
	@Override
	public boolean checkLength() throws MyException{
		if(str.length()<3)
			throw new MyException("I campi utilizzati per calcolare la password,\ndevono essere lunghi almeno 3 caratteri");
		return true;
	}
	@Override
	public void validate() throws MyException{
		//tolgo spazi
		str=str.replaceAll("\\s+","");
		checkLength();//meglio controllare la lunghezza dopo aver tolto gli spazi
		//controllo che non ci siano numeri o caratteri speciali
		if(!str.matches("[a-zA-Z]+")){
			throw new MyException("Parola non valida,\nsono consentiti solo numeri");
		}
	}

}